src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -112,6 +115,23 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     datasetToUpdate && datasetToUpdate.items.length === selectedIds.size
   )
 
+  useEffect(() => {
```

Comment:
I wonder if it would make sense to have a custom hook that encapsulates this behavior, looking something like

```tsx
function useOnChange<T>(
    obj: T,
    onChangeFn: (obj: T, prevObj: T) => void,
    comparator: (obj: T, prevObj: T) => boolean
  ) {
  const [previousObj, setPreviousObj] = useState(obj)
  useEffect(() => {
    if (!comparator(previousObj, obj)) {
      onChangeFn(obj, previousObj)
    }
    setPreviousObj(obj)
  }, [obj])
}
```

I don't think we need it now (I am not sure if we have this sort of behavior anywhere else right now).

What are your thoughts?

Reply:
I don't think I've seen this behavior else where either. Do you think this would be used in future cases?

Reply:
Not sure, let's just leave it for now